internal/cow/cow.go

```diff
@@ -0,0 +1,38 @@
+package cow
+
+import (
```

Comment:
Can you add some documentation on lifetimes for these methods. For example, what is the result of calling `ExitCode` on a running process? Does it wait or return error?

Reply:
All implementations will need to make sure to implement the same is why I want it written down

Reply:
Yep makes sense. Updated.

Reply:
test